ui/hooks/useDisplayName.ts

```diff
@@ -50,17 +55,21 @@ export function useDisplayNames(
 
     const name =
       nameEntry?.name ||
+      nftCollectionNameFromMetadata?.name ||
```

Comment:
It's not likely to ever happen, but should we be prioritising first party names over this?

Reply:
Done https://github.com/MetaMask/metamask-extension/pull/25692/commits/6907abeee82d8814535fc3e1a630cef644cb8c05